cheroot/connections.py

```diff
@@ -0,0 +1,254 @@
+                # No new connection, but reuse existing socket.
+                conn = socket_dict[rlist.pop()]
+            else:
+                conn = server_socket
+
+            # All remaining connections in rlist should be added
+            # to the ready queue.
+            if rlist:
+                socket_dict.update({
+                    fno: (socket_dict[fno], True)
+                    for fno in rlist
+                })
+
+            # New connection.
+            if conn is server_socket:
+                return self._from_server_socket(server_socket)
+
+        del self.connections[conn]
+        return conn
+
+    def _close(self, conn):
+        del self.connections[conn]
+        conn.close()
+
+    def _from_server_socket(self, server_socket):
+        try:
+            s, addr = server_socket.accept()
+            if self.server.stats['Enabled']:
+                self.server.stats['Accepts'] += 1
+            prevent_socket_inheritance(s)
+            if hasattr(s, 'settimeout'):
+                s.settimeout(self.server.timeout)
+
+            mf = MakeFile
+            ssl_env = {}
+            # if ssl cert and key are set, we try to be a secure HTTP server
+            if self.server.ssl_adapter is not None:
+                try:
+                    s, ssl_env = self.server.ssl_adapter.wrap(s)
+                except errors.NoSSLError:
+                    msg = (
+                        'The client sent a plain HTTP request, but '
+                        'this server only speaks HTTPS on this port.'
+                    )
+                    buf = [
+                        '%s 400 Bad Request\r\n' % self.server.protocol,
+                        'Content-Length: %s\r\n' % len(msg),
+                        'Content-Type: text/plain\r\n\r\n',
+                        msg,
+                    ]
+
+                    sock_to_make = s if six.PY3 else s._sock
+                    wfile = mf(sock_to_make, 'wb', io.DEFAULT_BUFFER_SIZE)
+                    try:
+                        wfile.write(''.join(buf).encode('ISO-8859-1'))
+                    except socket.error as ex:
+                        if ex.args[0] not in errors.socket_errors_to_ignore:
+                            raise
+                    return
+                if not s:
+                    return
+                mf = self.server.ssl_adapter.makefile
+                # Re-apply our timeout since we may have a new socket object
+                if hasattr(s, 'settimeout'):
+                    s.settimeout(self.server.timeout)
+
+            conn = self.server.ConnectionClass(self.server, s, mf)
+
+            if not isinstance(self.server.bind_addr, six.string_types):
+                # optional values
+                # Until we do DNS lookups, omit REMOTE_HOST
+                if addr is None:  # sometimes this can happen
+                    # figure out if AF_INET or AF_INET6.
+                    if len(s.getsockname()) == 2:
```

Comment:
How about:
```suggestion
                    addr = '::', 0
                    if len(s.getsockname()) == 2:
                        addr[0] = '0.0.0.0'
```

Reply:
I only moved this existing code from `server.py` to `connections.py`, so I don't want to get involved with making changes to that code (at least, not in this pull request).